src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -438,6 +439,20 @@ void DirectoryService::StartFirstTxEpoch()
         SetState(MICROBLOCK_SUBMISSION);
         m_dsStartedMicroblockConsensus = false;
 
+        if (BROADCAST_GOSSIP_MODE)
+        {
+            std::vector<Peer> peers;
+            for (const auto& i : *m_mediator.m_node->m_myShardMembers)
+            {
+                if (i.second.m_listenPortHost != 0)
+                {
+                    peers.push_back(i.second);
```

Comment:
Not critical, but let's use emplace_back more frequently instead of push_back from now on.